src/dmd/lexer.d

```diff
@@ -202,6 +203,104 @@ unittest
     }
 }
 
+
+/**************
+ * Pool allocator for Token objects, used by the Lexer.
+ * Tokens are never freed so Token pointers will never change
+ */
+private struct TokenPool
+{
+    private Array!Token tokens;
+
+    public ref Token getByIndex(size_t index) nothrow pure
```

Comment:
Perhaps use operator overloading instead making the following syntax possible: `pool[index]`.